openreview/venue/process/invitation_edit_process.py

```diff
@@ -37,47 +37,39 @@ def delete_invitation(child_invitation, ddate):
         )
     
     def get_children_notes():
-        source = invitation.content.get('source', {}).get('value', 'all_submissions') if invitation.content else False
-        reply_to = invitation.content.get('reply_to', {}).get('value', 'forum') if invitation.content else False
-        source_submissions_query = invitation.content.get('source_submissions_query', {}).get('value') if invitation.content else ''
-
-        print('source', source)
-        print('reply_to', reply_to)
-        print('source_submissions_query', source_submissions_query)
-        if source == 'accepted_submissions':
-            source_submissions = client.get_all_notes(content={ 'venueid': venue_id }, sort='number:asc', details='replies')
-            if not source_submissions and decision_name:
-                under_review_submissions = client.get_all_notes(content={ 'venueid': submission_venue_id }, sort='number:asc', details='replies')
-                source_submissions = [s for s in under_review_submissions if len([r for r in s.details['replies'] if f'{venue_id}/{submission_name}{s.number}/-/{decision_name}' in r['invitations'] and openreview.tools.is_accept_decision(r['content'][decision_field_name]['value'], accept_options) ]) > 0]
-        else:
-            source_submissions = client.get_all_notes(content={ 'venueid': submission_venue_id }, sort='number:asc', details='replies')
+        source = openreview.tools.get_invitation_source(invitation, domain)
+
+        ## TODO: use tools.should_match_invitation_source when "all_submissions" is removed
+        def filter_by_source(source):
+
+            venueids = source.get('venueid', [submission_venue_id]) ## we should always have a venueid
+            source_submissions = client.get_all_notes(content={ 'venueid': ','.join([venueids] if isinstance(venueids, str) else venueids) }, sort='number:asc', details='replies')
+
+            ## Keep backward compatibility with 'all_submissions' before and after running the post_decision_stage
             if not source_submissions:
                 source_submissions = client.get_all_notes(content={ 'venueid': ','.join([venue_id, rejected_venue_id]) }, sort='number:asc', details='replies')
+            
+            if 'with_decision_accept' in source:
+                source_submissions = [s for s in source_submissions 
+                                      if len([r for r in s.details['replies'] 
+                                        if f'{venue_id}/{submission_name}{s.number}/-/{decision_name}' in r['invitations'] 
+                                        and openreview.tools.is_accept_decision(r['content'][decision_field_name]['value'], accept_options) == source.get('with_decision_accept')]) > 0]
```

Comment:
@melisabok about the issue I found, I think the tests are passing because we are not using `should_match_invitation_source` in `invitation_edit_process.py` and here we are checking that the decision and `with_decision_accept` match.

Reply:
ok, but it is called in the process function of decision process for example so I will add a test to reproduce it

Reply:
I don't think it's being called there either? looks like we call it only in the submission, desk-rejection and withdrawal processes?